### PR TITLE
Add volumes and env vars to helm hook test pod

### PR DESCRIPTION
**Pull request by Iristyle**
- Fix test typo
 - Add basic server-test Pod tests 
     - This covers all existing functionality that matches what's
        present in server-statefulset.bats
 - Fix server-test helm hook Pod rendering 
     - Properly adhere to the global.enabled flag and the presence of
   the injector.externalVaultAddr setting, the same way that
   the servers StatefulSet behaves
 - Uses the same extraEnvironmentVars, volumes and volumeMounts set on
   the server statefulset to configure the Vault server test pod used by
   the helm test hook
 - This is necessary in situations where TLS is configured, but the
   certificates are not affiliated with the k8s CA / part of k8s PKI

 - Fixes GH-665

### CODE REVIEW

File: templates/tests/server-test.yaml [hunk 1]
```diff
@@ -35,6 +37,9 @@ spec:
           fi
 
           exit 0
-
+      volumeMounts:
+        {{ template "vault.mounts" . }}
```
Comment (Iristyle) [inline templates/tests/server-test.yaml:41]: This drags in the `audit` and `data` volumes which doesn't make sense... shooting for something simpler

**Comment by Iristyle** [comment]
I've verified this fixes the TLS issues in the test chart. Logs now look like:

```
Checking for sealed info in 'vault status' output                                                                                 
Attempt 0...                                                                                                                      
Error checking seal status: Get "https://cd4pe-vault.default.svc:8200/v1/sys/seal-status": dial tcp 10.106.187.188:8200: connect: 
Attempt 1...                                                                                                                      
Error checking seal status: Get "https://cd4pe-vault.default.svc:8200/v1/sys/seal-status": dial tcp 10.106.187.188:8200: connect: 
Attempt 2...                                                                                                                      
Error checking seal status: Get "https://cd4pe-vault.default.svc:8200/v1/sys/seal-status": dial tcp 10.106.187.188:8200: connect: 
Attempt 3...                                                                                                                      
Error checking seal status: Get "https://cd4pe-vault.default.svc:8200/v1/sys/seal-status": dial tcp 10.106.187.188:8200: connect: 
Attempt 4...                                                                                                                      
sealed: true
```

**Review by tvoran — COMMENTED** [review]
Would you mind adding unit tests for these options? I think it's probably time we added a `test/unit/server-test.bats`. Should be able to use server-statefulset.bats for examples of testing `extraEnvironmentVars`, `volumeMounts`, and `volumes`.

File: templates/tests/server-test.yaml [hunk 2]
```diff
@@ -35,6 +36,13 @@ spec:
...
           exit 0
-
+      volumeMounts:
+        {{- if .Values.server.volumeMounts }}
+          {{- toYaml .Values.server.volumeMounts | nindent 12}}
+        {{- end }}
+  volumes:
+    {{- if .Values.server.volumes }}
+      {{- toYaml .Values.server.volumes | nindent 8}}
+    {{- end }}
```
[the remark this answers is not in the document]
Comment (Iristyle) [reply templates/tests/server-test.yaml:46]: I can re-render and check... but I also did verify that the current indentation works in a live deployment to map the volumes in question.

[the remark this answers is not in the document]
Comment (Iristyle) [reply templates/tests/server-test.yaml:48]: You were right on the indentation rendering wrong -- verified with

> helm template --show-only templates/tests/server-test.yaml --set 'server.volumes[0].name=plugins' --set 'server.volumes[0].emptyDir=\{\}' --set 'server.volumeMounts[0].name=plugins' --set 'server.volumeMounts[0].mountPath=/usr/local/libexec/vault' --set 'server.volumeMounts[0].readOnly=true' .

**Comment by Iristyle** [comment]
Follow-on PR at #678 brings some more feature parity to the test pod

File: test/unit/server-test.bats [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+@test "server/standalone-server-test-Pod: enable with server.standalone.enabled true" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      --show-only templates/tests/server-test.yaml  \
+      --set 'server.standalone.enabled=true' \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
```
Comment (tvoran) [inline test/unit/server-test.bats:57]: Would you mind adding one more test like this, but for "server.ha.enabled true"?

**Review by tvoran — COMMENTED** [review]
Thanks for adding all the tests. Just one more suggestion and I think we'll be good to go.

File: test/unit/server-test.bats (hunk 3 above)
Comment (Iristyle) [reply test/unit/server-test.bats:57]: Done

**Review by tvoran — APPROVED** [review]
Thanks!

**Pull request merged**